src/cloudant/index.py

```diff
@@ -90,8 +95,48 @@ class Index(object):
     Instantiated with the raw data callable such as the
     CloudantDatabase.all_docs or View.__call__ reference used to
     retrieve data, the index can also store optional extra args for
-    customisation and supports efficient, paged iteration over the
-    results to avoid large views blowing up memory
+    customisation and supports efficient, paged iteration over
+    the view to avoid large views blowing up memory
+
+    In python, slicing returns by value, wheras iteration will yield
+    elements of the sequence, which means that using slices is better
+    for smaller slices of data, wheras if you have large views
+    its better to iterate over them as it should be more efficient.
+
+    Examples:
+
+    Access by key:
+    index['key'] # get all records matching key
+
+    Slicing by startkey/endkey
+
+    index[["2013","10"]:["2013","11"]] # results between compound keys
+    index["2013":"2014"] # results between string keys
+    index["2013":] # all results after key
+    index[:"2014"] # all results up to key
```

Comment:
My brain is too muzzy today to be of much in terms of critique. But being able to do slices on a couch view really is awesome -- this doc string is exciting to read :-) +1, would read again.